_posts/2019-05-10-server-rendering-responsively.md

```diff
@@ -0,0 +1,189 @@
+much about that history.)
+
+We've also built our site using responsive design, so you get a browsing experience optimized for your device.
+
+Combining SSR and responsive design is a non-trivial problem. There are many concerns to manage, and they are
+sometimes in conflict with each other. We server render for performance reasons, but we also want to be sure our
+app is performant when your browser takes over, all while optimizing accessibility and SEO.
+
+This article describes the tools we use on [artsy.net](https://artsy.net) to combine SSR and responsive design.
+
+<!-- more -->
+
+## Tool 1: [`styled-system`](https://styled-system.com)
+
+We handle the majority of responsive styling differences with
+[`styled-system`](https://styled-system.com/responsive-styles). This has been a really great addition to our
+toolbox. Here's a component that would render a `div` (`Box`) with a width of 50% at our mobile breakpoint, 75% at
+our tablet breakpoint, and 100% for anything larger:
+
+```xml
+<Box width={["50%", "75%", "100%"]}>
+  ...
+</Box>
+```
+
+As developers, we love this experience. We can make subtle differences to components across breakpoints with very
+little code and effort.
+
+We're working on another blog post that digs deeper into how we use `styled-system` within
+[our design system](https://palette.artsy.net/). You can also
+[poke around our source](https://github.com/artsy/reaction/blob/f3dabb884d616c5c42bbd47b1432dc2a9456b8ca/src/Apps/Search/Components/SearchResultsSkeleton/Header.tsx#L6)
+to see how much we've embraced `styled-system`'s responsive styles. (In that example, `pr` means `padding-right`
+and `pl` means `padding-left`.)
+
+There's one type of challenge with building a responsive app that `styled-system` can't solve: when we need to emit
+different layouts across different breakpoints. In this case, we need something that can render very different
+component sub-trees. We couldn't find an approach that satisfied our needs, so we wrote our own.
+
+## Tool 2: [`@artsy/react-responsive-media`](https://github.com/artsy/react-responsive-media)
+
+First off, an announcement: we've just released `react-responsive-media` version 1.0! (TODO: release v1.0 😬 or
+change this announcement)
+
+`react-responsive-media` allows you to define a set of breakpoint widths, then declaratively render component
+sub-trees when those breakpoints are met. It looks something like this:
+
+```xml
+<>
+  <Media at="xs">
+    <MobileLayout />
+  </Media>
+  <Media greaterThan="xs">
+    <NonMobileLayout />
+  </Media>
+</>
+```
+
+In this example, we're emitting the `MobileLayout` component for devices at or below our `xs` breakpoint, and the
+`NonMobileLayout` for devices greater than our `xs` breakpoint. You can imagine that the `MobileLayout` and
+`NonMobileLayout` components are complicated sub-trees, with more significant differences than `styled-system`
```

Comment:
```
You can imagine that the `MobileLayout` and `NonMobileLayout` components [contain] complicated sub-trees
```

Reply:
@damassi I'd recommend using suggestions to make this easier to act on. 

![image](https://user-images.githubusercontent.com/3087225/57643860-be497d00-7588-11e9-961b-16789f79ab69.png)

It's the +/- button.

Reply:
Ahh, will to 👍